libraries/botbuilder-adapter-alexa/src/alexaAdapter.ts

```diff
@@ -0,0 +1,150 @@
+import {
+    BotAdapter,
+    ConversationReference,
+    TurnContext,
+    Activity,
+    ResourceResponse,
+    WebRequest,
+    WebResponse,
+    ActivityTypes
+} from 'botbuilder';
+import { RequestEnvelope, IntentRequest, Session, interfaces, SessionEndedRequest } from 'ask-sdk-model';
+import { getRequestType, getLocale, getUserId, createAskSdkError } from 'ask-sdk-core';
+import { retrieveBody } from '@botbuildercommunity/adapter-twitter';
+
+/**
+ * @module botbuildercommunity/adapter-alexa
+ */
+
+export class AdapterAlexa extends BotAdapter {
+    private responses: Map<string, Activity[]> = new Map<string, Activity[]>();
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async continueConversation(_reference: Partial<ConversationReference>, _logic: (revocableContext: TurnContext) => Promise<void>): Promise<void> { }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async deleteActivity(_context: TurnContext, _reference: Partial<ConversationReference>): Promise<void> {
+        throw new Error(`deleteActivity is not implemented for ${ AdapterAlexa.name }`);
+    }
+
+    public async sendActivities(context: TurnContext, activities: Partial<Activity>[]): Promise<ResourceResponse[]> {
+        const resourceResponses: ResourceResponse[] = [];
+        const key = this.createKey(context);
+        this.responses.set(key, activities as Activity[]);
+        for (const activity of activities) {
+            resourceResponses.push({
+                id: activity.id || ''
+            });
+        }
+
+        return resourceResponses;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public async updateActivity(_context: TurnContext, _activity: Partial<Activity>): Promise<void> {
+        throw new Error(`updateActivity is not implemented for ${ AdapterAlexa.name }`);
+    }
+
+    public async processActivity(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+        const alexaRequest: RequestEnvelope = await retrieveBody(req);
+        const system: interfaces.system.SystemState = alexaRequest.context.System;
+        const requestType = getRequestType(alexaRequest);
+        const session: Session = alexaRequest.session ? alexaRequest.session : {
```

Comment:
session is something in the alexa sdk. in the .net adapter, they use the session to add an `id` property to the conversation. my guess is so that the `conversation.id` is consistent between alexa and botframework.